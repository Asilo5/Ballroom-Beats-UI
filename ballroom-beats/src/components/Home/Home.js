import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';
import Icon from 'react-native-vector-icons/AntDesign';

class Home extends Component {

  render() {
      return (
          <View style={styles.container}>
            <Text style={styles.homeHeader}>Ballroom Beatz</Text>
            <Icon.Button 
                style={styles.button} 
                name="arrowright" 
                onPress={() => this.props.navigation.navigate('DanceType')} 
                title="ENTER">
                <Text style={styles.enter}>ENTER</Text>
            </Icon.Button>
          </View>
      )      
  }
} 

const AppNavigator = createStackNavigator({
    Home: {
      screen: Home,
    },
});
  
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'black',
        alignItems: 'center',
        justifyContent: 'center',
    },
    homeHeader: {
        fontSize: 40,
        color: 'white',
        marginBottom: 25,
  },
    button: {
        backgroundColor: '#32CD32',
        fontSize: 25,
    },
    enter: {
        fontSize: 25,
        fontWeight: 'bold',
    }
});

export default withNavigation(Home);