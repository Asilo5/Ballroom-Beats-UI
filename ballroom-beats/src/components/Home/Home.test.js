import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home/Home';

const mockProps = {
   navigation: {
     navigate: jest.fn(),
     getParam: jest.fn(),
   }
 };

describe('Home', () => {

   it('should match snapshot', () => {
    const home = renderer.create(<Home navigation={mockProps.navigation} />).toJSON();
    expect(home).toMatchSnapshot();
   });

});
