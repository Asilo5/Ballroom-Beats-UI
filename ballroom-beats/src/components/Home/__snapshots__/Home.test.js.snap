// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#666363",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 450,
        "resizeMode": "stretch",
        "width": 425,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#007AFF",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "#007AFF",
            "borderRadius": 5,
          },
          Object {
            "backgroundColor": "#A9C344",
          },
        ]
      }
      title="ENTER"
    >
      <Text />
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        ENTER
      </Text>
    </View>
  </View>
</View>
`;
